python/src/aiconfig/schema.py

```diff
@@ -884,6 +906,7 @@ def _update_model_for_aiconfig(
 `update_model` function with a specified `prompt_name` argument.
 """
         warnings.warn(warning_message)
+        model_name = model_name.strip()
```

Comment:
If a config already has model name with leading or trailing space here, it can no longer have settings set by this?

Reply:
Since aiconfig-level model settings can have more than a single model, if we change the model name, we have no idea what the "previous" model name used to be so will error if we're only trying to pass in settings. Therefore, if you change the model name and don't pass in settings argument, it will just create a new model with empty settings. I wrote a comment about this in https://github.com/lastmile-ai/aiconfig/blob/43c7f9ea1592f1e00f812f2c75111475d4100f39/python/src/aiconfig/schema.py#L784-L788